### PR TITLE
Tweek a few of the archiving methods to better support BridgeSDK.

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**

### CODE REVIEW

File: Research/Research/RSDDataArchiveManager.swift [hunk 1]
```diff
@@ -340,7 +349,7 @@ internal class TaskArchiver : NSObject {
         // Look to see if the result conforms to the archivable protocol or the collection
         // protocol. If it conforms to both, then *only* archive it at this level and do not
         // recurse into the result.
-        if let achivable = result as? RSDArchivable {
+        if let achivable = archive.archivableData(for: result, sectionIdentifier: sectionIdentifier, stepPath: stepPath) {
```
Comment (Erin-Mounts) [inline Research/Research/RSDDataArchiveManager.swift]: achivable --> archivable (?)

File: Research/Research/RSDTaskPath.swift [hunk 2]
```diff
@@ -50,6 +50,10 @@ public final class RSDTaskPath : NSObject, NSCopying {
     /// and the app needs to track what the scheduled timing is for the task.
     public var scheduleIdentifier: String?
     
+    /// The user info is a pointer to task data that is required to run this task. This is a
+    /// pointer so it can be anything needed by the task.
```
Comment (Erin-Mounts) [inline Research/Research/RSDTaskPath.swift]: I'm confused about referring to this as a pointer. Was it adapted from Obj-C code? Does the comment need to be updated?

**Review by Erin-Mounts — APPROVED** [review]
couple of nits, otherwise looks good

File: Research/Research/RSDTaskPath.swift (hunk 2 above)
Comment (syoung-smallwisdom) [reply Research/Research/RSDTaskPath.swift]: No, originally it was an `Any`. Then I thought that the Apple pattern is `userInfo` as a dictionary, but really, I don't see why it needs to be. Basically, I am looking for a name for a thing that can hold a reference to a `clientData` object.

**Pull request merged**